Solutions/Marain.UserNotifications.Hosting.AspNetCore/Microsoft/Extensions/DependencyInjection/UserNotificationsHostServiceCollectionExtensions.cs

```diff
@@ -145,11 +149,33 @@ public static class UserNotificationsHostServiceCollectionExtensions
         public static IServiceCollection AddOpenApiJsonSerializerSettings(
             this IServiceCollection services)
         {
-            // Ideally we'd have a fully customised setup here rather than using the Corvus extension method and then
-            // removing what we don't want. However, it won't be possible to do that until
-            // https://github.com/corvus-dotnet/Corvus.Extensions.Newtonsoft.Json/issues/129 is resolved.
-            services.AddJsonSerializerSettings();
-            services.Remove(services.First(x => x.ImplementationType == typeof(DateTimeOffsetConverter)));
+            services.AddJsonNetSerializerSettingsProvider();
+            services.AddJsonNetPropertyBag();
+            services.AddJsonNetCultureInfoConverter();
+            services.AddSingleton<JsonConverter>(new StringEnumConverter(true));
+
+            return services;
+        }
+
+        /// <summary>
+        /// Ensures that the DateTimeOffsetConverter from Corvus.Extensions.Newtonsoft.Json is not present in the
+        /// service collection.
+        /// </summary>
+        /// <param name="services">The target service collection.</param>
+        /// <returns>The service collection.</returns>
+        /// <remarks>
+        /// Various of the Marain components add this by default, primarily now for backwards compatibility. However,
+        /// this service has never used the DateTimeOffsetConverter, so we need to make sure it's not in the service
+        /// collection. This method should be called at the end of your service initialisation.
```

Comment:
This baffles me. How have we ended up introducing a change in behaviour (adding the `DateTimeOffsetConverter`) in the name of backwards compatibility if it results in a change in behaviour?

Reply:
So, this comes down to the fact that we used to register all of the converters in the single method.

Throughout the different areas of code, there were various places where that method was called. And the method determined whether or not it should run based on the presence of a single item in the service collection. That meant that it didn't matter where our call to the registration method appeared, because once that method had been called it would never get called again, despite the fact that we'd removed one of the things it added.

However, now the code we're using is calling the individual methods, they all check for the specific thing they are adding. So if we remove the `DateTimeOffsetConverter` early on in the registration process, it's quite possible that something else will add it back in later in the process.

Ideally we'd stop any of the Marain bits adding the `DateTimeOffsetConverter` but this would take us into the territory of the updated tenancy service not being backwards compatible, so that's a change for the future.

Reply:
So we've managed to pave the way for doing what we wanted, but we're not yet ready to do it.. Fair enough.